### DISCUSSION TITLE
npm annotations are producing wrong results

### DISCUSSION THREAD

**Discussion by lmilbaum** [renovatebot/renovate#22339] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

In my dockerfile:
```
# renovate: datasource=node depName=node
ARG NODE_VERSION=18.16.0

# renovate: datasource=node depName=npm versioning=npm
ARG NPM_VERSION=9.6.6

# renovate: datasource=node depName=n versioning=npm
ARG N_VERSION=9.1.0

# renovate: datasource=npm depName=renovate versioning=npm
ARG RENOVATE_VERSION=35.98.0

# renovate: datasource=npm depName=yarn versioning=npm
ARG YARN_VERSION=1.22.19
```

Renovate generated 2 MRs so far:
```
# renovate: datasource=node depName=n versioning=npm
ARG N_VERSION=9.1.0
ARG N_VERSION=v18.16.0
```

```
# renovate: datasource=node depName=npm versioning=npm
ARG NPM_VERSION=9.6.6
ARG NPM_VERSION=v18.16.0
```

The `v` prefix should no be there and the version is wrong. It picks up the node versions instead.

### What you tried so far.

Reading the documentation and looking for inspiration from other projects.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment]
Looks like you need to use `extractVersion`

**Reply by lmilbaum** [reply]
Thanks @rarkins. I've just found out the problem. I've used the wrong `datasource` value. It worked as soon as I changed it `npm`.
